src/vellum/workflows/events/utils.py

```diff
@@ -1,5 +1,62 @@
+from optparse import Option
+from typing import Optional
+
+from vellum.workflows.events.types import (
+    NodeParentContext,
+    ParentContext,
+    PromptDeploymentParentContext,
+    WorkflowDeploymentParentContext,
+    WorkflowParentContext,
+)
 from vellum.workflows.events.workflow import WorkflowEvent
+from vellum_ee.workflows.display.vellum import CodeResourceDefinition
 
 
 def is_terminal_event(event: WorkflowEvent) -> bool:
-    return event.name in {"workflow.execution.fulfilled", "workflow.execution.rejected", "workflow.execution.paused"}
+    return event.name in {
+        "workflow.execution.fulfilled",
+        "workflow.execution.rejected",
+        "workflow.execution.paused",
+    }
+
+
+def set_current_context(current_context_json: dict) -> Optional[ParentContext]:
+    type_dict = {
+        "WORKFLOW": WorkflowParentContext,
+        "WORKFLOW_NODE": NodeParentContext,
+        "WORKFLOW_RELEASE_TAG": WorkflowDeploymentParentContext,
+        "PROMPT_RELEASE_TAG": PromptDeploymentParentContext,
+    }
+    current_type_value = current_context_json.get("type", "DEFAULT")
+    current_type = type_dict.get(current_type_value, None)
+    if current_type:
+        temp = current_type(**current_context_json)
+        return temp
+    else:
+        return None
+
+
```

Comment:
helper util to convert json to parent_context classes (ParentContext serializers does *not* like building)